### PR TITLE
Add plugin installer from http archive

### PR DESCRIPTION
**Pull request by jlyheden**
Our use case is that we want to package plugins as tarballs and be able to install them via the helm cli by referencing the url to the plugin artifact.

### CODE REVIEW

**Review by bacongobbler — COMMENTED** [review]
Thanks for the change! Would you mind writing up some unit tests and documentation to cover this new feature? Thanks!

**Comment by jlyheden** [comment]
Added unit tests that cover update and failure scenarios in addition to the happy installation path and a small section in the plugin docs how to install from url, let me know if anything else is needed.

**Review by technosophos — APPROVED** [review]
With the tests, this all seems sane to me. Sorry for the long delay getting this second review going.

**Review by bacongobbler — APPROVED** [review]

**Pull request merged**
